### PR TITLE
[NTOS:CC] Raise status, when CcMapData fails

### PR DESCRIPTION
**Pull request by tkreuzer**
Also add a check in fastfat, that asserts, when CcMapData returns FALSE.

### CODE REVIEW

File: ntoskrnl/cc/pin.c [hunk 1]
```diff
@@ -427,7 +427,9 @@ CcMapData (
             if (iBcb == NULL)
             {
                 CcRosReleaseVacb(SharedCacheMap, Vacb, TRUE, FALSE, FALSE);
+                *pBcb = NULL; // If you ever remove this for compat, make sure to review all callers for using an unititialized value
```
Comment (HBelusca) [inline ntoskrnl/cc/pin.c:430]: 😆

**Comment by zefklop** [comment]
I'm in the process of rewriting all of this. Please hold your horses before merging

**Comment by SergeGautherie** [comment]
What is left of this PR, after switch to fastfat_new?

**Comment by zefklop** [comment]
This is still needed.

**Pull request merged**